packages/next/build/webpack/loaders/next-flight-server-loader.ts

```diff
@@ -208,7 +212,7 @@ export default async function transformSource(
   }
 
   const isServerComponent = createServerComponentFilter(extensions)
-  const isClientComponent = createClientComponentFilter(extensions)
+  const isClientComponent = createClientComponentFilter()
```

Comment:
What’s the reason behind this?

Reply:
client components can have any extension, not only the page extensions. you can have `a.client.mjs` but it is not a valid page extension yet

Reply:
Yeah, so in that sense server components can have any extension too. We should do the same for both component types, but only apply the `pageExtension` check for page entries.